src/exporters/certHelpers.go

```diff
@@ -35,41 +37,66 @@ func secondsToExpiryFromCertAsBase64String(s string) ([]certMetric, error) {
 	return secondsToExpiryFromCertAsBytes(certBytes)
 }
 
+func decodeFromPKCS(certBytes []byte) ([]*pem.Block, error) {
+	var blocks []*pem.Block
+	pfx_blocks, err := pkcs12.ToPEM(certBytes, "")
+	if err != nil {
+		return nil, err
+	}
+	for _ , b := range pfx_blocks {
+		if b.Type == "CERTIFICATE" {
+			blocks = append(blocks, b)
+		}
+	}
+	return blocks, nil
+}
+
+
 func secondsToExpiryFromCertAsBytes(certBytes []byte) ([]certMetric, error) {
 	var metrics []certMetric
 	var blocks []*pem.Block
+	var last_err error
 
 	// Export the first certificates in the certificate chain
 	block, rest := pem.Decode(certBytes)
-	if block == nil {
-		return metrics, fmt.Errorf("Failed to parse as a pem")
-	}
-	blocks = append(blocks, block)
-
-	// Export the remaining certificates in the certificate chain
-	for len(rest) != 0 {
-		block, rest = pem.Decode(rest)
-		if block == nil {
-			return metrics, fmt.Errorf("Failed to parse intermediate as a pem")
+	if block != nil {
+		blocks = append(blocks, block)
+		// Export the remaining certificates in the certificate chain
+		for len(rest) != 0 {
+			block, rest = pem.Decode(rest)
+			if block == nil {
+				return metrics, fmt.Errorf("Failed to parse intermediate as a pem")
+			}
+			if block.Type == "CERTIFICATE" {
+				blocks = append(blocks, block)
+			}
 		}
-		if block.Type == "CERTIFICATE" {
-			blocks = append(blocks, block)
+	} else {
+		// Parse as PKCS ?
+		pfx_blocks, err := decodeFromPKCS(certBytes)
+		if err != nil {
+			return metrics, fmt.Errorf("Failed to parse certificate")
```

Comment:
let's include an indication that there was an attempt to parse as both pem and the new format in the error message for clarity.

also let's include the reported error. something like:
`fmt.Errorf("failed to parse as pem and pkcs12: %w", err)`